src/codeflare_sdk/job/jobs.py

```diff
@@ -108,15 +110,58 @@ def _dry_run(self, cluster: "Cluster"):
             workspace=f"file://{Path.cwd()}",
         )
 
-    def submit(self, cluster: "Cluster") -> "Job":
+    def _missing_spec(self, spec: str):
+        raise ValueError(f"Job definition missing arg: {spec}")
+
+    def _dry_run_no_cluster(self):
+        return torchx_runner.dryrun(
+            app=ddp(
+                *self.script_args,
+                script=self.script,
+                m=self.m,
+                name=self.name if self.name is not None else self._missing_spec("name"),
```

Comment:
total nit, do you wan this on one line like this instead of
```
name=self.name
if self.name is not None
else self._missing_spec("name")

```

Reply:
Ah, I just run `black` to auto-format, guessing that line wasn't deemed "too long", but I can make that uniform with the rest